### PR TITLE
Testing and coverage

### PR DESCRIPTION
**Pull request by stigrj**
Set up Travis CI and codecov

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+matrix:
+  include:
+    - os: linux
+      addons:
+        apt:
+          packages:
+            - cmake3
+            - cmake3-data
+            - g++-4.8
+            - gfortran-4.8
```
Comment (robertodr) [inline .travis.yml]: Why is the dreaded Fortran needed?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            - libeigen3-dev
+            - libboost-math-dev
+      env:
+        - BUILD_TYPE='release'
+        - CXX_COMPILER='g++-4.8'
+        - TESTS='--enable-tests'
```
Comment (robertodr) [inline .travis.yml]: I think you'll always want tests to run on travis. I'd hardwire this flag into the setup line.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  - cmake --version
+  - ${CXX_COMPILER} --version
+
+script:
+  - export OMP_NUM_THREADS=2
+  - ./setup --type=${BUILD_TYPE} --cxx=${CXX_COMPILER} ${TESTS} ${COVERAGE} ${OPENMP}
+  - cd build
+  - cmake --build . -- VERBOSE=1
+  - |
+    if [[ "${TESTS}" = "--enable-tests" ]]; then
```
Comment (robertodr) [inline .travis.yml]: If you always enable tests this is redundant.

File: .travis.yml (hunk 1 above)
Comment (stigrj) [reply .travis.yml]: Just trying to get a successful build, xcfun asks for a fortran compiler: https://travis-ci.org/stigrj/mrchem/builds/352748045

**Comment by stigrj** [comment]
@robertodr do you think we can use the same encrypted slack notification token for mrchem as we already have for mrcpp?

**Comment by robertodr** [comment]
I would think not.

**Pull request merged**